### DISCUSSION TITLE
Reloading rotated certificate with Keycloak on Quarkus

### DISCUSSION THREAD

**Discussion by tsaarni** [keycloak/keycloak#10654] [Archive]
Is it possible to "hot-reload" renewed/rotated TLS certificate and key without restarting Keycloak?  The question applies to all interfaces, for example HTTPS server certificate or client certificate towards backend database such as Postgres.

Previously it was possible to reload using WildFly CLI.

**Comment by tsaarni** [comment]
I believe these two issues need to be resolved to support the feature for HTTPS server certificate:

- https://github.com/eclipse-vertx/vert.x/issues/3780
- https://github.com/quarkusio/quarkus/issues/15926

Client certificate for Postgres database connection is reloaded at every connection attempt already as of today.

**Comment by ecerulm** [comment]
Do you know if it's possible currently in keycloak 22.0.x? 

If i understood right from https://github.com/quarkusio/quarkus/issues/15926#issuecomment-1545002471 it should be possible to hot reload a TLS certificate in Quarkus already today, but does Keycloak 22 takes advantage of that possibility?

**Comment by tsaarni** [comment]
No, Keycloak does not have this functionality yet.  The standard JDK KeyManager is still used, and it does not reload when certificates change on disk.

Quarkus [makes it possible](https://github.com/quarkusio/quarkus/pull/27682) for an application to set own KeyManager - which allows application to provide an alternative implementation that can hot-reload certificates. I've [worked with this topic](https://github.com/keycloak/keycloak-community/pull/345) a bit since posting this, but there has not been consensus on the feature yet.

**Comment by tsaarni** [comment]
I've converted this to an issue https://github.com/keycloak/keycloak/issues/26524
